app/src/main/java/me/ivg2/foodapp/RecipeFragment.java

```diff
@@ -86,6 +95,59 @@ public void onClick(View view) {
         rvRecipes.setLayoutManager(new LinearLayoutManager(getActivity()));
         rvRecipes.setAdapter(recipeAdapter);
 
+        tabs = view.findViewById(R.id.tab_host);
+        tabs.setSelectedTabIndicatorColor(getResources().getColor(R.color.colorAccent));
+        tabs.setTabTextColors(Color.parseColor("#000000"),getResources().getColor(R.color.colorAccent));
+        tabs.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
+            @Override
+            public void onTabSelected(TabLayout.Tab tab) {
+                refreshRecommendedRecipes();
+
+                switch (tab.getPosition()) {
+                    case 0:
+                        RecipeItemRepository.set(cookingRecipes);
+                        recipeAdapter.notifyDataSetChanged();
+                        rvRecipes.smoothScrollToPosition(0);
+                        tab.getIcon().setColorFilter(getResources().getColor(R.color.colorAccent), PorterDuff.Mode.SRC_IN);
+                        return;
+                    case 1:
+                        ArrayList<Recipe> tempRecipes = new ArrayList<>();
+                        for (int j = 0; j < 5; j++) {
+                            for (int i = 0; i < browsingRecipes.size(); i++) {
+                                if (browsingRecipes.get(i).getIngredientsMissing().size() == j) {
+                                    tempRecipes.add(browsingRecipes.get(i));
+                                }
+                            }
+                        }
+                        browsingRecipes = tempRecipes;
+                        //cookingRecipes = RecipeItemRepository.getAll();
+                        RecipeItemRepository.set(browsingRecipes);
+                        recipeAdapter.notifyDataSetChanged();
+                        rvRecipes.smoothScrollToPosition(0);
+                        tab.getIcon().setColorFilter(getResources().getColor(R.color.colorAccent), PorterDuff.Mode.SRC_IN);
+                        return;
+                }
+            }
+
+            @Override
+            public void onTabUnselected(TabLayout.Tab tab) {
+                switch (tab.getPosition()) {
+                    case 0:
+                        tab.getIcon().setColorFilter(Color.parseColor("#000000"), PorterDuff.Mode.SRC_IN);
+                        return;
+                    case 1:
+                        tab.getIcon().setColorFilter(Color.parseColor("#000000"), PorterDuff.Mode.SRC_IN);
+                        return;
+                }
+            }
+
+            @Override
+            public void onTabReselected(TabLayout.Tab tab) {
```

Comment:
Do we need this funciton if it is empty?

Reply:
We need to implement the functions from the abstract class TabLayout.OnTabSelectedListener(). But I don't need anything in this one.